src/fu-daemon.c

```diff
@@ -65,10 +69,40 @@ struct _FuDaemon {
 
 G_DEFINE_TYPE(FuDaemon, fu_daemon, G_TYPE_OBJECT)
 
+#define FU_DAEMON_HOUSEKEEPING_DELAY 10 /* seconds */
+
+static gboolean
+fu_daemon_schedule_housekeeping_cb(gpointer user_data)
+{
+	FuDaemon *self = FU_DAEMON(user_data);
+
+#ifdef HAVE_MALLOC_TRIM
+	/* drop heap except one page */
+	malloc_trim(0);
```

Comment:
0?  Shouldn't be 4096?

Reply:
The man page is quite unclear: * The  pad  argument  specifies  the amount of free space to leave untrimmed at the top of the heap.  If this argument is 0, only the minimum amount of memory is maintained at the top of the heap (i.e., one page or less).  A nonzero argument can be used to maintain some trailing space at the top of the heap in order to allow future allocations to be made without having to  extend the heap with sbrk(2).*

I wasn't sure if `free space` was in pages or bytes. 0 seemed to avoid the problem altogether.

Reply:
Got it makes sense.